### ISSUE TITLE
Implement stock loss/adjustment native form

### ISSUE THREAD

**Issue by cagulas** [OpenSRP/opensrp-client#482]
- [x] Tapping the back button or the cancel button should open the form close confirmation message. If confirmed, the user should go back to the current stock screen for that vaccine #479 
- [x] The fields in the form should follow the fields defined in the data dictionary spreadsheet, see the "Stock Control" tab
- [x] Form styling uses same library as birth registration form
- [x] Tapping "Confirm" submits the form and updates the table in the current stock screen for that vaccine #479
- [x] Can do error states if the user tries to click "Confirm" and required fields aren't all completed. If we are using the same library as birth registration, we can treat confirm button on this form the same way.
- [x] All of the above actions should be supported both online and offline
- [x] Important note - the number they enter for stock loss/adj should not allow the balance to become a negative number.
- [x] This form uses a number picker widget. It doesn't have to look exactly like the below, but some criteria are important:
- it starts at 0
- ideally the number itself is editable with keypad in addition to just using the minus/plus buttons
- numbers can be negative or positive

![227432472](https://cloud.githubusercontent.com/assets/3488986/24952771/68b5c3b2-1f81-11e7-8d9a-6e0294f49c51.png)

**Comment by cagulas** [comment]
@raihan-mpower Here are my suggestions:

- [ ] The word "stock" is missing in the label "Date of [vaccine] stock loss/adjustment"
- [ ] Upper constraint is missing on the date of loss/adjustment. User should not be able to put a date more than 1 calendar month ago.
- [ ] Physical recount and Other options are WAY too big!
- [ ] "Please specify" field is missing if "Other" is selected.
- [ ] Remove space before colon in "New balance" field.
- [ ] **The app is constantly crashing in this form. Seems to be when I'm tapping on the plus or minus buttons on the number of vials to be adjusted or trying to type in a value.**
- [ ] If the user taps in the text area for vials adjustment, the alpha keypad appears. It  would be nice if the numeric keypad appeared instead (note: it should allow negative numbers).
- [ ] Vials [vaccine] adjustment should have an asterisk, as it's a required field.
- [ ] Fix the error message shown if balance is negative number, i.e. white font and don't capitalize "Make" and add period at end of sentence. 
- [ ] Is it possible to default the Vials [vaccine] adjustment field to be 0 when the user opens the form?
- [ ] There seems to be an extra space before the vaccine name in the Date of [vaccine] loss/adjustment field.
- [ ] In all loss/adjustment forms, it is saying M/MR in the reason for adjustment field instead of the correct vaccine name.

**Comment by cagulas** [comment]
@raihan-mpower Thanks for the fixes. Here are a few things I noticed:

- [ ] Remove space before colon in "New balance" field. Doesn't seem to be done.
- [ ] The app is still crashing. 
- [ ] Something seems to have broken and it's allowing negative balances now in this form. I was able to type in a negative number and submit the form. The stock shows a negative balance now. 
- [ ] As a result, I wasn't able to test "Fix the error message shown if balance is negative number, i.e. white font and don't capitalize "Make" and add period at end of sentence."
